cpp/tests/copying/scatter_tests.cpp

```diff
@@ -60,13 +60,11 @@ TEST_F(ScatterUntypedTests, ScatterScalarMapNulls)
 {
   using cudf::scalar_type_t;
   using cudf::test::fixed_width_column_wrapper;
-  using scalar_ptr    = std::unique_ptr<cudf::scalar>;
-  using scalar_vector = std::vector<scalar_ptr>;
 
-  // Initializers lists can't take move-only types
-  scalar_vector source_vector;
-  auto source = scalar_ptr(new scalar_type_t<int32_t>(100));
-  source_vector.push_back(std::move(source));
+  std::vector<std::reference_wrapper<const cudf::scalar>> source_vector;
+  const cudf::scalar* source_slr                         = new scalar_type_t<int32_t>(100);
+  std::reference_wrapper<const cudf::scalar> slr_wrapper = std::ref(*source_slr);
+  source_vector.push_back(slr_wrapper);
```

Comment:
Use this same pattern throughout. 
```suggestion
  auto const source = scalar_type_t<int32_t>{100};
  std::vector<std::reference_wrapper<const cudf::scalar>> source_vector{std::cref(source)};
```

Reply:
👍 will do.

Reply:
I updated this, but the compiler isn't happy for some reason - looking into it.

Reply:
@jrhemstad I'm sure that this is just due to a little lack of context for the language, but it's not really obvious to me what is going wrong here. Since a morning of poking around google and SO didn't turn anything up, would you mind taking a quick look at the compiler errors and letting me know if anything sticks out to you?